### ISSUE TITLE
Train segmentation model on MS-COCO things classess only

### ISSUE THREAD

**Issue by Hrayo712** [open-mmlab/mmsegmentation#1658]
Dear,
I have a couple of questions I hope you could help me clarifying:

**1)**  I'm interested in training a model only on the things classes of COCO-stuff. I see that this is not supported out-of-the-box, as the official support is for COCO-stuff (things + classess). But I wonder if there is maybe an easy way in which I can configure the dataset to just focus on the things classes ?

**2)** Assuming it is possible to train on the things classes only, is there an easy way in which I can configure training to only train on a subset of the classes ? say, only 20 out of the 80 ?

Thanks a lot for the help

Regards

**Comment by MengzhangLI** [comment]
You can use MMDetection to fulfill your instance/ panoptic segmentation requirements of MS-COCO dataset.
MMSegmentation only supports semantic segmentation.

**Comment by Hrayo712** [comment]
Thanks for your reply. I believe I did not explained myself properly, so let me rephrase:

As far as I understand, coco-stuff is supported as a dataset on mmsegmentation, and can be used for training a semantic segmentation models. That being said, I am only interested on the things classes of the coco-stuff dataset (coco-stuff has 172 classes: 80 thing classes, 91 stuff classes ). So my question was about an easy way to remove/mask/configure the stuff classes so that the model trains only on the things classess.

I hope this is clearer now. I apologize for the confusion

Thanks!

**Comment by MengzhangLI** [comment]
OK, got it. I think it may need to re-generate brand new coco-stuff dataset.

First, in dataset preparation step, design a dict which only has things: https://github.com/open-mmlab/mmsegmentation/blob/master/tools/convert_datasets/coco_stuff164k.py#L14-L187. Current `len(trID)` is 171 (0 to 170), perhaps your requirement is 80 things label, so it is necessary to find out what `clsID` is you want and convert things only while omitting stuff label. 

Then renew `CLASSES` and `PALETTE` in [COCOStuffDataset](https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/datasets/coco_stuff.py#L17-L90) class. There are only newly 80 (label 0 to 79) labels.

What's more, `num_classes` in config file should also be modified, from `171` to `80`.

Hope my answer could help you.
